src/cmd/ksh93/sh/jobs.c

```diff
@@ -434,7 +438,11 @@ bool job_reap(int sig) {
         }
 #endif  // SHOPT_COSHELL
         pid = waitpid((pid_t)-1, &wstat, flags);
-        sh_offstate(shp, SH_TTYWAIT);
+        if (was_ttywait_on) {
```

Comment:
If it was already on there isn't any need to enable it again, is there? Shouldn't this just be

```
if (!was_ttywait_on) sh_offstate(shp, SH_TTYWAIT);
```

Reply:
Also, there is a `break` inside the loop so it seems like the same statement is needed around line 580 when the loop exits.